Reduce 'android-18' duplication by putting it in $ANDROID_TARGET env variable

mitchhentges opened the pull request:
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy --faster` does not report any errors
- [x] These changes fix #8348

Either:
- [X] These changes do not require tests because ~~I'm lazy~~ it's a configuration change, and asserting default values is usually tedious and not very helpful

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process.

---

Allegedly, this broke the build [back in the day](https://github.com/servo/servo/pull/8519).
Going to see if it's a piece of cake to land now

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/11278)

<!-- Reviewable:end -->

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @metajack (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/command_base.py

mitchhentges commented on support/android/apk/project.properties:
```diff
@@ -1 +0,0 @@
-target=android-18
```
I'm going a little ballsy here, and assuming that the contents of `support/android/apk` are built with the variable specified [here](https://github.com/servo/servo/pull/11278/files#diff-40f5a7cf22f94aad059b2c1795347f5eR104).

Notice that in [`build.xml`](https://github.com/servo/servo/pull/11278/files#diff-2419f69b360f1fa393abc46d9d331d01R39), `project.properties` is no longer loaded, and `target` is assumed from the environment variable `$ANDROID_TARGET`

aneeshusa commented:
Queue is empty and this looks reasonable. @bors-servo try

bors-servo commented:
:hourglass: Trying commit d6995bc with merge 4b17640...

bors-servo commented:
:broken_heart: Test failed - [android](http://build.servo.org/builders/android/builds/993)

aneeshusa commented:
Looks like the same error as last time. I'll try to repro this locally tomorrow with the servo-linux-android1 VM from [saltfs](https://github.com/servo/saltfs/), which has the same configuration as the android Buildbot builder.

mitchhentges commented:
Ah, figured out why it's failing:
[`openssl.sh` sets up a `_ANDROID_API` variable](https://github.com/servo/servo/blob/master/support/android/openssl.sh#L39), which is then [used later in the same script](https://github.com/servo/servo/blob/master/support/android/openssl.sh#L128).
If  `_ANDROID_API` isn't set properly (say, if it's set to a blank environment variable), then `ANDROID_SYSROOT` is broken, and the build fails.

The underlying issue is that this PR addresses the `.servobuild`, machine-local case of specifying `$ANDROID_TARGET`. However, it doesn't address the "default" case, when no `target` is explicitly specified.

Where should such default configuration go? I can put "if blank then set to default" logic in each of the ~5 places I've changed in this PR, but it's a bad solution and it would make it tedious to change the `target` default in the future.

highfive commented:
New code was committed to pull request.

mitchhentges commented:
@bors-servo try

mitchhentges commented:
(I wanted to believe)

highfive commented:
New code was committed to pull request.

mitchhentges commented:
I've now set `$ANDROID_PLATFORM` to default to `android-18` in `command_base.py`.
Additionally, I'm using `android.platform` instead of `android.target` because that was already used (`arm-linux-androideabi`)

jdm commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit 3067fee with merge b3b2658...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/1132)

highfive commented:
▶ TIMEOUT [expected OK] /html/dom/reflection-embedded.html
      │ 
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.14/src/lib.rsthread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;:thread &#39;873http_loader for http://web-platform.test:8000/html/dom/test-valueOf
      │ &#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, stack backtrace:
      │ could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ ../src/libstd/rand/mod.rs:159
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050cc9bf - rand::thread_rng::THREAD_RNG_KEY::__init::h4e1387a2e4a21393
      │    9:      thread &#39; http_loader for http://web-platform.test:8000/html/dom/test-valueOf &#39; panicked at &#39;0xcould not initialize thread_rng: Too many open files (os error 24)1050cc7fc&#39;,  - /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.14/src/lib.rs:rand873::
      │ thread_rng::he394b446ab605489
      │   10:        0x104ad5ee2 - uuid::Uuid::new_v4::hc206acac9c7e580d
      │   11:        0x1038b9a8b - net::http_loader::load::h981055a57a1b11f1
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   12:        0xthread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │ &#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   13:        0x1050fe6bb - __rust_try
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.14/src/lib.rs:A&gt;&gt;::call_box::873ha5012a14880088fa
      │ 
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.14/src/lib.rs:873
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, /Users/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.14/src/lib.rs:873
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs   8:159
      │ :        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   13:        0x1050fe6bb - __rust_try
      │ thread &#39;http_loader for http://web-platform.test:</span><span class="stdout">8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │ thread &#39;http_loader for http://web-platform.test:8000/html/dom/test-valueOf&#39; panicked at &#39;could not initialize thread_rng: Too many open files (os error 24)&#39;, ../src/libstd/rand/mod.rs:159
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050cc9bf - rand::thread_rng::THREAD_RNG_KEY::__init::h4e1387a2e4a21393
      │    9:        0x1050cc7fc - rand::thread_rng::he394b446ab605489
      │   10:        0x104ad5ee2 - uuid::Uuid::new_v4::hc206acac9c7e580d
      │   11:        0x1038b9a8b - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      �</span><span class="stdout">�    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050cc9bf - rand::thread_rng::THREAD_RNG_KEY::__init::h4e1387a2e4a21393
      │    9:        0x1050cc7fc - rand::thread_rng::he394b446ab605489
      │   10:        0x104ad5ee2 - uuid::Uuid::new_v4::hc206acac9c7e580d
      │   11:        0x1038b9a8b - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050cc9bf - rand::thread_rng::THREAD_RNG_KEY::__init::h4e1387a2e4a21393
      │    9:        0x1050cc7fc - rand::thread_rng::he394b446ab605489
      │   10:        0x104ad5ee2 - uuid::Uuid::new_v4::hc206acac9c7e580d
      │   11:        0x1038b9a8b - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6f</span><span class="stdout">e4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050cc9bf - rand::thread_rng::THREAD_RNG_KEY::__init::h4e1387a2e4a21393
      │    9:        0x1050cc7fc - rand::thread_rng::he394b446ab605489
      │   10:        0x104ad5ee2 - uuid::Uuid::new_v4::hc206acac9c7e580d
      │   11:        0x1038b9a8b - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2</span><span class="stdout">b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      │   18:     0x7fff95827fd6 - _pthread_start
      │ stack backtrace:
      │    1:        0x1050f4aa8 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
      │    2:        0x1050faf35 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
      │    3:        0x1050fab4e - std::panicking::default_hook::hc2c969e7453d080c
      │    4:        0x104889bf2 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::ha2bd86c312dc8d7a
      │    5:        0x1050e2702 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
      │    6:        0x1050fb4f6 - std::panicking::begin_panic::h4889569716505182
      │    7:        0x1050e3ff8 - std::panicking::begin_panic_fmt::h484cd47786497f03
      │    8:        0x1050fb969 - std::rand::thread_rng::THREAD_RNG_KEY::__init::h536e3917997a1628
      │    9:        0x1050e2d0c - std::rand::thread_rng::h789d00d1fce4f79b
      │   10:        0x10387a0ac - _&lt;collections..hash..map..RandomState as core..default..Default&gt;::default::ha847494db5044892
      │   11:        0x1038b8dc0 - net::http_loader::load::h981055a57a1b11f1
      │   12:        0x1038b4a62 - std::panicking::try::call::h2f55588c0568c703
      │   13:        0x1050fe6bb - __rust_try
      │   14:        0x1050fe655 - __rust_maybe_catch_panic
      │   15:        0x1038b6fe4 - _&lt;F as std..boxed..FnBox&lt;A&gt;&gt;::call_box::ha5012a14880088fa
      │   16:        0x1050f9f58 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
      │   17:     0x7fff95828059 - _pthread_body
      └   18:     0x7fff95827fd6 - _pthread_start

mitchhentges commented:
I don't think that the failing test is related to my changes, though I may be wrong

aneeshusa commented on support/android/build-apk/src/main.rs:
```diff
@@ -90,14 +90,18 @@ fn main() {
 
+    // Retrieve the Android SDK version target env variable
+    let default_platform = "android-18".to_owned();
+    let android_platform = env::var("ANDROID_PLATFORM").unwrap_or(default_platform);
```
Let's make passing the environment variable mandatory, just like the other ANDROID_* variables (see the top of main for examples).

aneeshusa commented:
@mitchhentges Looks good to me aside from the one review comment. Go ahead and squash while you're at it.
The test failure is a known intermittent: #11100.

-S-awaiting-review +S-needs-code-changes +S-needs-squash

highfive commented:
New code was committed to pull request.

highfive commented:
New code was committed to pull request.

aneeshusa commented:
The commit looks good; please clean up the commit message. The last two lines were fixup commit messages that are no longer needed, and we should use full sentences in the body of the message (not the title though).

highfive commented:
New code was committed to pull request.

mitchhentges commented:
I think I got it this time :)

Commit:
Reduce 'android-18' duplication, as originally done by wenderen

Use android.platform (and $ANDROID_PLATFORM) for configuration, apply 'android-18' default

highfive commented:
New code was committed to pull request.

aneeshusa commented:
Thanks! @bors-servo r+

bors-servo commented:
:pushpin: Commit 619212a has been approved by `aneeshusa`

<!-- @bors-servo r=aneeshusa 619212abf3d62a83e97dfdb3d8d1a085bf2db792 -->

bors-servo commented:
:hourglass: Testing commit 619212a with merge ea38ccf...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/1090), [arm32](http://build.servo.org/builders/arm32/builds/1158), [arm64](http://build.servo.org/builders/arm64/builds/1143), [linux-dev](http://build.servo.org/builders/linux-dev/builds/1160), [linux-rel](http://build.servo.org/builders/linux-rel/builds/1214), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/1194), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/1180), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/1222), [windows](http://build.servo.org/builders/windows/builds/868)
